amethyst_network/src/resources/net_event.rs

```diff
@@ -6,15 +6,14 @@ use shrev::Event;
 use resources::*;
 use serde::Serialize;
 use serde::de::DeserializeOwned;
-use specs::world::Component;
-use specs::storage::VecStorage;
+use specs::{VecStorage,Component};
 use uuid::Uuid;
 
-/// The basic network events shipped with amethyst
+/// The basic network events shipped with amethyst.
 // TODO: Add CreateEntity,RemoveEntity,UpdateEntity
 // TODO: Example of switching the NetEvent set in the Network Systems
 #[derive(Debug,Clone,PartialEq,Serialize,Deserialize)]
-pub enum NetEvent<T>{// where T:Send+Sync+Serialize+Clone+DeserializeOwned+'static{
+pub enum NetEvent<T> where T: PartialEq{//where T:Send+Sync+Serialize+Clone+DeserializeOwned+'static{
```

Comment:
Don't put bounds onto types where possible.
It isn't possible if:
1. Some field type has this bound.
2. You use associated type of the trait implementation.

Reply:
fixed